File: src/Services/ContributorService.js
import axios from "axios";

const CONTRIBUTOR_API_BASE_URL = "http://localhost:9003/contributor";

class ContributorService {
    
    register(contributor) {
        return axios.post(CONTRIBUTOR_API_BASE_URL + "/register", contributor);
    }

    login(contributor) {
        return axios.post(CONTRIBUTOR_API_BASE_URL + "/login", contributor);
    }

    addCourse(username, course) {
        return axios.post(CONTRIBUTOR_API_BASE_URL + "/" + username + "/course/add", course);
    }

    getAllCourses(username) {
        return axios.get(CONTRIBUTOR_API_BASE_URL + "/" + username + "/course/all");
    }

    editCourse(username, id, course) {
        return axios.put(CONTRIBUTOR_API_BASE_URL + "/" + username + "/course/edit/" + id, course);
    }

    deleteCourse(username, id) {
        return axios.delete(CONTRIBUTOR_API_BASE_URL + "/" + username + "/course/delete/" + id);
    }

    answerQuestion(username, pid, answer) {
        return axios.post(CONTRIBUTOR_API_BASE_URL + "/" + username + "/course/doubt/" + pid + "/answer", answer);
    }
}

export default new ContributorService();

File: backend/contributorservice/src/main/java/com/lg/contributor/services/ContributorService.java
package com.lg.contributor.services;

import java.util.List;

import com.lg.contributor.dto.AllCoursesResponse;
import com.lg.contributor.dto.AllDoubtsResponse;
import com.lg.contributor.dto.AnswerDoubtRequest;
import com.lg.contributor.dto.AuthRequest;
import com.lg.contributor.dto.ContributorSignupResponse;
import com.lg.contributor.dto.CourseAddRequest;
import com.lg.contributor.dto.CourseAddResponse;
import com.lg.contributor.dto.CourseEditRequest;
import com.lg.contributor.dto.CourseEditResponse;
import com.lg.contributor.dto.UserCoursesResponse;
import com.lg.contributor.model.Contributor;
import com.lg.contributor.model.Course;
import com.lg.contributor.model.Doubt;

public interface ContributorService {
	ContributorSignupResponse createContributor(Contributor contributor);
	Contributor login(AuthRequest authRequest);
	boolean removeCourse(String username, Integer id);
	CourseAddResponse addCourse(String username, CourseAddRequest car);
	CourseEditResponse editCourse(String username, Integer id, CourseEditRequest cer);
	List<UserCoursesResponse> getUserCourses(String username);
	List<AllCoursesResponse> getAllCourses();
	Course getCourseById(Integer id);
	String getContributorNameById(Integer id);
	List<Doubt> getAllDoubts();
	Doubt answerQuestion(Integer pid, AnswerDoubtRequest adr);
	List<AllDoubtsResponse> getAllAnswers();
}
